centreon/src/Core/Service/Infrastructure/Repository/DbReadServiceRepository.php

```diff
@@ -325,7 +325,18 @@ private function getServiceRequestConcatenator(array $accessGroupIds = []): SqlC
             );
 
         if ([] !== $accessGroupIds) {
-            // TODO: will be handled later
+            $concatenator->appendJoins(
+                <<<'SQL'
+                    JOIN `:dbstg`.centreon_acl acl
+                        ON s.service_id = acl.service_id
+                    SQL
+            );
+            $concatenator->appendWhere(
+                <<<'SQL'
+                    WHERE acl.group_id IN (:access_group_ids)
+                    SQL
+            );
+            $concatenator->storeBindValueMultiple(':access_group_ids', $accessGroupIds, \PDO::PARAM_INT);
```

Comment:
If a non-administrator user has no access groups, no ACLs will be used?

Reply:
I added the check in the existsByAccessGroups() public method